apps/tenant-management-webapp/src/app/pages/admin/services/comment/topicTypes/topicTypes.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from 'react';
+
+import { GoAButton } from '@abgov/react-components-new';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { getCommentTopicTypes, updateCommentTopicType } from '@store/comment/action';
+import { RootState } from '@store/index';
+import { renderNoItem } from '@components/NoItem';
+import { CommentTopicTypesTable } from './definitionsList';
+import { PageIndicator } from '@components/Indicator';
+import { defaultCommentTopicType } from '@store/comment/model';
+
+import { DeleteModal } from '@components/DeleteModal';
+import { AddEditCommentTopicType } from './addEditCommentTopicType';
+
+interface CommentTopicTypesProps {
+  openAddTopicTypes: boolean;
+}
+export const CommentTopicTypes = ({ openAddTopicTypes }: CommentTopicTypesProps) => {
+  const [showDeleteConfirmation, setShowDeleteConfirmation] = useState(false);
+  const [currentDefinition, setCurrentDefinition] = useState(defaultCommentTopicType);
+
+  const commentDefinitions = useSelector((state: RootState) => {
+    return Object.entries(state?.comment?.topicTypes)
+      .sort((template1, template2) => {
+        return template1[1].name.localeCompare(template2[1].name);
+      })
+      .reduce((tempObj, [commentDefinitionId, commentDefinitionData]) => {
+        tempObj[commentDefinitionId] = commentDefinitionData;
+        return tempObj;
+      }, {});
+  });
+
+  const [openAddCommentTopicType, setOpenAddCommentTopicType] = useState(false);
+
+  const indicator = useSelector((state: RootState) => {
+    return state?.session?.indicator;
+  });
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (openAddTopicTypes) {
+      setOpenAddCommentTopicType(true);
+    }
+  }, [openAddTopicTypes]);
+
+  useEffect(() => {
+    dispatch(getCommentTopicTypes());
+  }, []);
+
+  const reset = () => {
+    setOpenAddCommentTopicType(false);
+  };
+
+  // eslint-disable-next-line
+  useEffect(() => {}, [commentDefinitions]);
+  useEffect(() => {
+    document.body.style.overflow = 'unset';
```

Comment:
I think we should create common function for this, even though it is a one liner to show the scrollbar. 
It is a couple of files now.

Reply:
ok, its fine then. Everything looks good.